osu.Game.Rulesets.Catch/Beatmaps/CatchBeatmapProcessor.cs

```diff
@@ -74,42 +74,43 @@ private void applyPositionOffsets()
 
         private void initialiseHyperDash(List<CatchHitObject> objects)
         {
-            // todo: add difficulty adjust.
-            double halfCatcherWidth = CatcherArea.CATCHER_SIZE * (objects.FirstOrDefault()?.Scale ?? 1) / CatchPlayfield.BASE_WIDTH / 2;
+            if (objects.Count == 0)
+                return;
 
-            int lastDirection = 0;
-            double lastExcess = halfCatcherWidth;
-
-            int objCount = objects.Count;
+            List<CatchHitObject> objectWithDroplets = new List<CatchHitObject>();
 
-            for (int i = 0; i < objCount - 1; i++)
+            foreach (var currentObject in objects)
             {
-                CatchHitObject currentObject = objects[i];
-
-                // not needed?
-                // if (currentObject is TinyDroplet) continue;
+                if (currentObject is Fruit)
+                    objectWithDroplets.Add(currentObject);
+                if (currentObject is JuiceStream)
+                    foreach (var currentJuiceElement in currentObject.NestedHitObjects)
+                        if (!(currentJuiceElement is TinyDroplet))
+                            objectWithDroplets.Add((CatchHitObject)currentJuiceElement);
+            }
 
-                CatchHitObject nextObject = objects[i + 1];
+            double halfCatcherWidth = CatcherArea.CATCHER_SIZE * objectWithDroplets[0].Scale / CatchPlayfield.BASE_WIDTH / 2;
```

Comment:
This is being done in multiple locations now. Expose a static method from `CatcherArea`, `GetCatcherSize(BeatmapDifficulty difficulty)` and pass `Beatmap.BeatmapInfo.BaseDifficulty` available here in.

Moving forward, we need to keep catcher-specific calculations within `CatcherArea`/`Catcher`.